float64 theta
float64 phi
float64 psi
float64 theta_dot
float64 phi_dot
float64 psi_dot